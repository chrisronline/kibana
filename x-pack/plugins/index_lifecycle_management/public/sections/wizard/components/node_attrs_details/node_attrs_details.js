/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import {
  EuiFlyoutBody,
  EuiFlyoutFooter,
  EuiFlyout,
  EuiTitle,
  EuiInMemoryTable,
  EuiSpacer,
  EuiButtonEmpty,
  EuiCallOut,
} from '@elastic/eui';

export class NodeAttrsDetails extends PureComponent {
  static propTypes = {
    fetchNodeDetails: PropTypes.func.isRequired,
    close: PropTypes.func.isRequired,

    details: PropTypes.array,
    selectedNodeAttrs: PropTypes.string.isRequired,
  };

  componentWillMount() {
    this.props.fetchNodeDetails(this.props.selectedNodeAttrs);
  }

  render() {
    const { selectedNodeAttrs, details, close } = this.props;

    return (
      <EuiFlyout onClose={close}>
        <EuiFlyoutBody>
          <EuiTitle>
            <h2>
              Below is a list of nodes that contain the attribute: `{
                selectedNodeAttrs
              }`
            </h2>
          </EuiTitle>
          <EuiSpacer size="s" />
          <EuiCallOut
            style={{ marginTop: '1rem' }}
            title="Heads up"
            color="warning"
          >
            Be aware that the nodes listed here only directly match the node
            attribute string and other nodes might be affected by this policy
            due to other allocation rules.
          </EuiCallOut>
          <EuiSpacer size="s" />
          <EuiInMemoryTable
            items={details}
            columns={[
              { field: 'nodeId', name: 'ID' },
              { field: 'stats.name', name: 'Name' },
              { field: 'stats.host', name: 'Host' },
            ]}
            pagination={true}
            sorting={true}
          />
        </EuiFlyoutBody>
        <EuiFlyoutFooter>
          <EuiButtonEmpty iconType="cross" onClick={close} flush="left">
            Close
          </EuiButtonEmpty>
        </EuiFlyoutFooter>
      </EuiFlyout>
    );
  }
}
