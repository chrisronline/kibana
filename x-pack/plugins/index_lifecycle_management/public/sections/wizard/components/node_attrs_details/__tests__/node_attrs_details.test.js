/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';
import { shallow } from 'enzyme';
import { NodeAttrsDetails } from '../node_attrs_details';

describe('NodeAttrsDetails', () => {
  it('should render normally', () => {
    const fetchNodeDetails = jest.fn();
    const component = shallow(
      <NodeAttrsDetails
        fetchNodeDetails={fetchNodeDetails}
        close={() => {}}
        selectedNodeAttrs="foo"
      />
    );

    expect(fetchNodeDetails).toHaveBeenCalledWith('foo');
    expect(component).toMatchSnapshot();
  });

  it('should render a warning for allocation rules', () => {
    const component = shallow(
      <NodeAttrsDetails
        fetchNodeDetails={() => {}}
        close={() => {}}
        selectedNodeAttrs="foo"
        allocationRules={{}}
      />
    );

    expect(component.find('EuiCallOut')).toMatchSnapshot();
  });
});
