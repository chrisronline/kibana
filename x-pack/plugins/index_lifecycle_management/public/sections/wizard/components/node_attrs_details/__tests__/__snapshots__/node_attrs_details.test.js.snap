// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeAttrsDetails should render a warning for allocation rules 1`] = `
<EuiCallOut
  color="warning"
  size="m"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
  title="Heads up"
>
  Be aware that this index template has existing allocation rules which will affect the list of nodes these indices can be allocated to.
</EuiCallOut>
`;

exports[`NodeAttrsDetails should render normally 1`] = `
<EuiPortal>
  <EuiFlyout
    onClose={[Function]}
    ownFocus={true}
    size="m"
  >
    <EuiFlyoutBody>
      <EuiTitle
        size="m"
      >
        <h2>
          Nodes that contain the attribute: \`
          foo
          \`
        </h2>
      </EuiTitle>
      <EuiSpacer
        size="s"
      />
      <EuiInMemoryTable
        columns={
          Array [
            Object {
              "field": "nodeId",
              "name": "ID",
            },
            Object {
              "field": "stats.name",
              "name": "Name",
            },
            Object {
              "field": "stats.host",
              "name": "Host",
            },
          ]
        }
        items={Array []}
        pagination={true}
        responsive={true}
        sorting={true}
      />
    </EuiFlyoutBody>
    <EuiFlyoutFooter>
      <EuiButtonEmpty
        color="primary"
        flush="left"
        iconSide="left"
        iconType="cross"
        onClick={[Function]}
        type="button"
      >
        Close
      </EuiButtonEmpty>
    </EuiFlyoutFooter>
  </EuiFlyout>
</EuiPortal>
`;
